pandas/core/frame.py

```diff
@@ -9934,12 +9934,16 @@ def _series_round(ser: Series, decimals: int) -> Series:
                 raise TypeError("Values in decimals must be integers")
             new_cols = list(_dict_round(self, decimals))
         elif is_integer(decimals):
-            # Dispatch to Series.round
-            new_cols = [_series_round(v, decimals) for _, v in self.items()]
+            # Dispatch to Block.round
+            return self._constructor(
+                self._mgr.round(decimals=decimals, using_cow=using_copy_on_write()),
+                index=self.index,
```

Comment:
Are index and columns here necessary? These information are tied to the manager

Reply:
Thanks, this slipped through the copy paste.